app/controllers/users_controller.rb

```diff
@@ -1,2 +1,30 @@
 class UsersController < ApplicationController
+  before_action :validate_session
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def follow
+    @user = User.find(params[:id])
+    @user.followers << current_user
```

Comment:
@idprietop what if for some reason the user is already a follower, will this code prevent the duplication?